File: src/code-samples/langchain/agents-guardrails.py
"""Agents docs: guardrails middleware example."""

# :snippet-start: agents-guardrails-py
from langchain.agents import create_agent
from langchain.agents.middleware import PIIMiddleware
from langchain.tools import tool


@tool
def search(query: str) -> str:
    """Search for a query and return a short summary."""
    return f"Search results for: {query}"


agent = create_agent(
    model="anthropic:claude-sonnet-4-6",
    tools=[search],
    middleware=[PIIMiddleware("email")],
)
# :snippet-end:


# :remove-start:
if __name__ == "__main__":
    assert agent is not None
    print("✓ agents guardrails sample compiles")
# :remove-end:

File: src/code-samples/langchain/agents-guardrails.ts
// Agents docs: guardrails middleware example.

// :snippet-start: agents-guardrails-js
import { createAgent, piiMiddleware, tool } from "langchain";
import * as z from "zod";

const search = tool(({ query }) => `Search results for: ${query}`, {
  name: "search",
  description: "Search for a query and return a short summary.",
  schema: z.object({ query: z.string() }),
});

const agent = createAgent({
  model: "anthropic:claude-sonnet-4-6",
  tools: [search],
  middleware: [piiMiddleware("email")],
});
// :snippet-end:

// :remove-start:
async function main() {
  void agent;
  console.log("✓ agents guardrails sample compiles");
}

main();
// :remove-end:
